app/src/pages/inside/testCaseLibraryPage/expandedOptions/folder/folder.jsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2025 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+
+import { ChevronDownDropdownIcon } from '@reportportal/ui-kit';
+import isEmpty from 'lodash.isempty';
+
+import styles from './folder.scss';
+
+const cx = classNames.bind(styles);
+
+export const Folder = ({ folder, setActiveFolder, activeFolder, setIsEmptyFolder }) => {
```

Comment:
I am afraid we might have challenges with this recursive-like rendering while adding keyboard interactions, accessibility, sync selection with url. Let's clarify our requirements.